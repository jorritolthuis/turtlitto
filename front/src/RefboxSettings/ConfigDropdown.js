import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import Dropdown from "../Dropdown";

const MODE_VALUES = ["Practice", "Dual team", "Match"];
const CONFIG_VALUES = [
  "Role assigner on",
  "Role assigner off",
  "Pass demo",
  "Penalty mode",
  "Ball Handling demo"
];

/**
 * Constructs two dropdowns for the type of game and role
 * Author: G.W. van der Heijden
 * Author: S.A. Tanja
 * Author: T.T.P. Franken
 *
 * Props:
 * - value: current value of the dropdown
 * - onChange: a function on what to do when the button is pressed
 */
const ConfigDropdown = props => {
  return (
    <DropDownSection>
      <RefboxDropdown
        currentValue={props.value}
        values={MODE_VALUES}
        onChange={value => {
          props.onChange("mode", value);
        }}
        enabled={true}
      />
      <RefboxDropdown
        currentValue={props.value}
        values={CONFIG_VALUES}
        onChange={value => {
          props.onChange("role", value);
        }}
        enabled={true}
      />
    </DropDownSection>
  );
};

const DropDownSection = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  justify-content: center;
`;

const RefboxDropdown = styled(Dropdown)`
  width: 75%;
  margin-left: auto;
  margin-right: auto;
  height: 4rem;
`;

ConfigDropdown.propType = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default ConfigDropdown;
