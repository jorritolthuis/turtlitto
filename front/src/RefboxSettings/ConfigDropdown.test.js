import React from "react";
import ConfigDropdown from "./ConfigDropdown";
import { shallow } from "enzyme";

describe("ConfigDropdown", () => {
  it("shows two dropdown menus", () => {
    const wrapper = shallow(<ConfigDropdown value="" onChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
