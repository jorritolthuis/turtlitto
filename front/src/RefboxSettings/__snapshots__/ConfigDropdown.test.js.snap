// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigDropdown shows two dropdown menus 1`] = `
<styled.div>
  <Styled(Dropdown)
    currentValue=""
    enabled={true}
    onChange={[Function]}
    values={
      Array [
        "Practice",
        "Dual team",
        "Match",
      ]
    }
  />
  <Styled(Dropdown)
    currentValue=""
    enabled={true}
    onChange={[Function]}
    values={
      Array [
        "Role assigner on",
        "Role assigner off",
        "Pass demo",
        "Penalty mode",
        "Ball Handling demo",
      ]
    }
  />
</styled.div>
`;
